test/integration/tracing/tracing_test.go

```diff
@@ -113,111 +108,77 @@ func TestTracing(t *testing.T) {
-			"failed to read emojivoto yaml\n%s\n", err)
-	}
-	emojivotoYaml = strings.ReplaceAll(emojivotoYaml, "___TRACING_NS___", tracingNs)
-	out, stderr, err := TestHelper.PipeToLinkerdRun(emojivotoYaml, "inject", "-")
-	if err != nil {
-		testutil.AnnotatedFatalf(t, "'linkerd inject' command failed",
-			"'linkerd inject' command failed\n%s\n%s", out, stderr)
-	}
-
-	out, err = TestHelper.KubectlApply(out, emojivotoNs)
-	if err != nil {
-		testutil.AnnotatedFatalf(t, "'kubectl apply' command failed",
-			"'kubectl apply' command failed\n%s", out)
-	}
-
-	// Ingress components
-	// Ingress must run in the same namespace as the service it routes to (web)
-	ingressNs := emojivotoNs
-	err = TestHelper.CreateDataPlaneNamespaceIfNotExists(ctx, ingressNs, nil)
-	if err != nil {
-		testutil.AnnotatedFatalf(t, fmt.Sprintf("failed to create %s namespace", ingressNs),
-			"failed to create %s namespace: %s", ingressNs, err)
-	}
-
-	ingressYaml, err := testutil.ReadFile("testdata/ingress.yaml")
-	if err != nil {
-		testutil.AnnotatedFatalf(t, "failed to read ingress yaml",
-			"failed to read ingress yaml\n%s\n", err)
-	}
-	ingressYaml = strings.ReplaceAll(ingressYaml, "___INGRESS_NAMESPACE___", ingressNs)
-	ingressYaml = strings.ReplaceAll(ingressYaml, "___TRACING_NS___", tracingNs)
-	out, stderr, err = TestHelper.PipeToLinkerdRun(ingressYaml, "inject", "-")
-	if err != nil {
-		testutil.AnnotatedFatalf(t, "'linkerd inject' command failed",
-			"'linkerd inject' command failed\n%s\n%s", out, stderr)
-	}
-
-	out, err = TestHelper.KubectlApply(out, ingressNs)
-	if err != nil {
-		testutil.AnnotatedFatalf(t, "'kubectl apply' command failed",
-			"'kubectl apply' command failed\n%s", out)
-	}
+	TestHelper.WithDataPlaneNamespace(ctx, "emojivoto", map[string]string{}, t, func(t *testing.T, emojivotoNs string) {
 
-	// wait for deployments to start
-	for _, deploy := range []struct {
-		ns   string
-		name string
-	}{
-		{ns: emojivotoNs, name: "vote-bot"},
-		{ns: emojivotoNs, name: "web"},
-		{ns: emojivotoNs, name: "emoji"},
-		{ns: emojivotoNs, name: "voting"},
-		{ns: ingressNs, name: "nginx-ingress"},
-		{ns: tracingNs, name: "collector"},
-		{ns: tracingNs, name: "jaeger"},
-	} {
-		if err := TestHelper.CheckPods(ctx, deploy.ns, deploy.name, 1); err != nil {
-			if rce, ok := err.(*testutil.RestartCountError); ok {
-				testutil.AnnotatedWarn(t, "CheckPods timed-out", rce)
-			} else {
-				testutil.AnnotatedError(t, "CheckPods timed-out", err)
-			}
+		emojivotoYaml, err := testutil.ReadFile("testdata/emojivoto.yaml")
+		if err != nil {
+			testutil.AnnotatedFatalf(t, "failed to read emojivoto yaml",
+				"failed to read emojivoto yaml\n%s\n", err)
+		}
+		emojivotoYaml = strings.ReplaceAll(emojivotoYaml, "___TRACING_NS___", tracingNs)
+		out, stderr, err := TestHelper.PipeToLinkerdRun(emojivotoYaml, "inject", "-")
+		if err != nil {
+			testutil.AnnotatedFatalf(t, "'linkerd inject' command failed",
+				"'linkerd inject' command failed\n%s\n%s", out, stderr)
```

Comment:
Similar point as above.

Can we add `linkerd.io/inject: enabled` to the manifests instead? We can apply them by calling into `kubectl` using the test helper.